api/src/main/java/io/jsonwebtoken/SignatureAlgorithm.java

```diff
@@ -113,6 +116,7 @@
     //purposefully ordered higher to lower:
     private static final List<SignatureAlgorithm> PREFERRED_HMAC_ALGS = Collections.unmodifiableList(Arrays.asList(
         SignatureAlgorithm.HS512, SignatureAlgorithm.HS384, SignatureAlgorithm.HS256));
+    private static final Map<String, SignatureAlgorithm> PREFERRED_HMAC_ALGS_LOOKUP = preferredHmacAlgorithmMap();
```

Comment:
Nit: Suggested name of `HMAC_ALGS_BY_JCA_ID` (and corresponding helper method)?  I tend to name lookup maps in this way to self-document the map keys (since 'lookup' tends to be generic').

Also, unless I'm missing something, we can drop the `PREFERRED` prefix because nothing about the map is used for preferred ordering checks, no?

Reply:
Agreed! I renamed that var three times locally trying to figure out a better name 😆